test/e2e/node_ip_migration.go

```diff
@@ -0,0 +1,680 @@
+				var err error
+				migrationWorkerNodeIP, err = findLastFreeSubnetIP(
+					externalContainerName,
+					externalContainerIPs[ipAddrFamily],
+					[]string{workerNodeIPs[ipAddrFamily], secondaryWorkerNodeIPs[ipAddrFamily], egressIP},
+				)
+				framework.Logf("New worker node IP will be %s", migrationWorkerNodeIP)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			JustAfterEach(func() {
+				if rollbackNeeded {
+					By(fmt.Sprintf("Migrating worker node %s back from IP address %s to IP address %s (inverted)",
+						workerNode.Name, migrationWorkerNodeIP, workerNodeIPs[ipAddrFamily]))
+					err := migrateWorkerNodeIP(workerNode.Name, migrationWorkerNodeIP, workerNodeIPs[ipAddrFamily],
+						true)
+					Expect(err).NotTo(HaveOccurred())
+
+					// It takes ovn-controller 200 seconds to reconnect to the SBDB after an IP address change.
+					// See here for more details: https://github.com/ovn-org/ovn-kubernetes/pull/3131#issuecomment-1351871875
+					By(fmt.Sprintf("Sleeping for %d seconds to give things time to settle", settleTimeout))
+					time.Sleep(time.Duration(settleTimeout) * time.Second)
+				}
+			})
+
+			Context("when no EgressIPs are configured", func() {
+				for _, updateKubeletFirst := range []bool{true, false} {
+					updateKubeletFirst := updateKubeletFirst // Required to avoid race conditions due to pointer assignment.
+					It(fmt.Sprintf("makes sure that the cluster is still operational (%s)",
+						updateKubeletIPAddressMsg[updateKubeletFirst]),
+						func() {
+							By(fmt.Sprintf("Migrating worker node %s from IP address %s to IP address %s (inverted)",
+								workerNode.Name, workerNodeIPs[ipAddrFamily], migrationWorkerNodeIP))
+							err := migrateWorkerNodeIP(workerNode.Name, workerNodeIPs[ipAddrFamily], migrationWorkerNodeIP,
+								true)
+							Expect(err).NotTo(HaveOccurred())
+
+							By("Setting rollbackNeeded to true")
+							rollbackNeeded = true
+
+							// It takes ovn-controller 200 seconds to reconnect to the SBDB after an IP address change.
+							// See here for more details: https://github.com/ovn-org/ovn-kubernetes/pull/3131#issuecomment-1351871875
+							By(fmt.Sprintf("Sleeping for %d seconds to give things time to settle", settleTimeout))
+							time.Sleep(time.Duration(settleTimeout) * time.Second)
+
+							By(fmt.Sprintf("Poking from pod %s on node %s to pod %s on node %s for a maximum of %d seconds",
+								podWorkerNode.Name, podWorkerNode.Spec.NodeName,
+								podSecondaryWorkerNode.Name, podSecondaryWorkerNode.Spec.NodeName,
+								pollingTimeout))
+							Eventually(func() bool {
+								By("Poking the pod")
+								err := pokeAllPodIPs(f, podWorkerNodeName, podSecondaryWorkerNode)
+								if err != nil {
+									framework.Logf("Poking all pod IPs failed: %q", err)
+								}
+								return err == nil
+							}, pollingTimeout, pollingInterval).Should(BeTrue())
+						})
+				}
+			})
+
+			Context("when EgressIPs are configured", func() {
```

Comment:
what is the relationship between egress IP and node ip address migration?

Reply:
noreroute LR policies will be updated when node IP changes.